### PR TITLE
Notmain

### PR DESCRIPTION
**Pull request by MaciejWdev**

### CODE REVIEW

File: alert/app.py [hunk 1]
```diff
@@ -14,6 +14,8 @@ def create_app():
 
     app.config.from_object("config.settings")
 
+    app.config["SECRET_KEY"] = "supersecretunguessablekey"
```
Comment (eugene-prout) [inline alert/app.py]: The Dynaconf loader sets the application secret key (from the config) to be the `SECRETKEY` value of the `app.config` dict. We want the two values to be the same. Can't think of a neat way fix this, so replace this line with `app.config["SECRET_KEY"] = app.config["SECRETKEY"]`.

File: alert/forms.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from flask import request
+from flask_wtf import FlaskForm
+from wtforms import TextAreaField, SubmitField
+from wtforms.validators import ValidationError, DataRequired, Length
+
+
+class EditGrammarForm(FlaskForm):
+    grammar_editor = TextAreaField("Enter Your Grammar", validators=[DataRequired()])
```
Comment (eugene-prout) [inline alert/forms.py:8]: Nice, the default validator doesn't look bad so let's keep this styling for now 👍 . Maybe add some more validation as we discussed but this is a nice expressive code.

File: alert/lib/fixed_grammer.bnf [hunk 3]
```diff
@@ -0,0 +1,102 @@
+program ::= extern_list decl_list
```
Comment (eugene-prout) [inline alert/lib/fixed_grammer.bnf]: Not a fan of these grammar files being here. I appreciate this isn't the final state but let's split these up and move them over to the tests folders when tests are added. For now, nothing to do.

File: alert/templates/application/components/main_panel.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        </ul>
+
+        <div class="tab-content" id="tabControl">
+            <div class="tab-pane active" id="editor-tab-pane" role="tabpanel"
+                aria-labelledby="editor-tab" tabindex="0">
+                <form action="" method="post">
+                    {{ form.hidden_tag() }}
+                    <p>
+                        {{ form.grammar_editor.label }}<br>
+                        {{ form.grammar_editor(cols=100,rows=25) }}<br>
```
Comment (eugene-prout) [inline alert/templates/application/components/main_panel.html:36]: I _think_ this is causing the formatting to break a little bit and not fit nicely on one page. This is fine for now, I'll make some changes to the formatting when this version is ready to go. For now, nothing, just an FYI

File: alert/lib/script.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    for item in non_terminals.values():
+        new_node = dot.node(item.name)
+
+    for item in non_terminals.values():
+        for dest in item.dependencies:
+            dot.edge(item.name, dest.name)
+        if item.top_level_terminals is not None:
+            dot.edge(item.name, item.top_level_terminals.get_rep())
+
+    dot.render(directory="doctest-output", view=True)
```
Comment (eugene-prout) [inline alert/lib/script.py]: When first reading this codebase I thought the `doctest` stuff was to do with testing documentation. (Tools exist to turn formatted documentation into test cases.) This is a confusing name in this line and other occurrences so it should be changed. I'd suggest  `output_graphs` or something similar.

File: alert/lib/script.py
Comment (eugene-prout) [inline alert/lib/script.py:1]: The functions and classes here are not my finest work. I'll take responsibility for this and fix this later. I'll neaten up the whole /lib folder and start out some tests which you can flesh out later if you fancy?

File: alert/lib/script.py (hunk 5 above)
Comment (eugene-prout) [inline alert/lib/script.py]: Changing line 113 to `dot.render(directory="doctest-output", view=True, format="jpg")` saves in the folder as a jpg making it easier to show the user. Not worried about this now, just an FYI

File: .gitignore [hunk 6]
```diff
@@ -93,6 +94,10 @@ dmypy.json
 # profiling data
 .prof
 
+# lib doctests
+alert/lib/doctest-output/
```
Comment (eugene-prout) [inline .gitignore:98]: Keeping the irrelevant files out of source control 👍

**Review by eugene-prout — CHANGES_REQUESTED** [review]
Overall good but needs two changes:
- Use the  `SECRETKEY` config var to sign the CSRF token
- Change the `doctest` name to something more descriptive

File: alert/lib/script.py
Comment (MaciejWdev) [reply alert/lib/script.py:1]: Sure! Any other guidance of how to get good test coverage? I'll probably end up asking this again if it still doesn't make sense after seeing a couple of your tests.

File: alert/lib/script.py (hunk 5 above)
Comment (MaciejWdev) [reply alert/lib/script.py]: I'll change this now so I don't forget later that this is likely the change needed to make work easier once we present the user their graph.

**Review by eugene-prout — APPROVED** [review]

**Pull request merged**